druid/src/tests/mod.rs

```diff
@@ -14,18 +14,42 @@
 
 //! Additional unit tests that cross file or module boundaries.
 
-mod harness;
+pub(crate) mod harness;
 mod helpers;
 mod layout_tests;
 
 use std::cell::Cell;
+use std::env;
+use std::fs;
 use std::rc::Rc;
+use tempfile;
 
 use crate::widget::*;
 use crate::*;
 use harness::*;
 use helpers::*;
 
+/// This function takes a closure and calls it will the path to a temperary direcotry
+///
+/// This is useful as after the closer has run the tempdir will be destoroyed
+/// so it will not interfear with any other tests or subsiquent involcations of
+/// the same test.
+#[allow(dead_code)]
+pub fn temp_dir_for_test() -> std::path::PathBuf {
+    let current_exe_path = env::current_exe().unwrap();
+    let mut exe_dir = current_exe_path.parent().unwrap();
+    if exe_dir.ends_with("deps") {
+        exe_dir = exe_dir.parent().unwrap();
+    }
+    let test_dir = exe_dir.parent().unwrap().join("tests");
+    fs::create_dir_all(&test_dir).unwrap();
+    tempfile::Builder::new()
+        .prefix("TempDir")
+        .tempdir_in(test_dir)
+        .unwrap()
+        .into_path()
```

Comment:
When does this cleanup happen? When the returned `PathBuf` gets dropped?

Reply:
Yep, sorry i have updated the docs to include this, as you pointed out the docs were for a first draft of this function.